### PR TITLE
Added batch instrumentation

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
fixes gh-1904

### CODE REVIEW

File: tests/brave/spring-cloud-sleuth-instrumentation-batch-tests/src/test/resources/application.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+logging.level.org.springframework.cloud: DEBUG
+logging.level.com.netflix.discovery.InstanceInfoReplicator: ERROR
+logging.level.org.springframework.cloud.sleuth.brave.instrument.web.client.feign: TRACE
+
+spring.autoconfigure.exclude: org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration, org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration, org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration, org.springframework.boot.autoconfigure.data.web.SpringDataWebAutoConfiguration, org.springframework.cloud.gateway.config.GatewayAutoConfiguration, org.springframework.cloud.gateway.config.GatewayClassPathWarningAutoConfiguration, org.springframework.cloud.gateway.config.GatewayMetricsAutoConfiguration
```
Comment (cppwfs) [inline tests/brave/spring-cloud-sleuth-instrumentation-batch-tests/src/test/resources/application.yml:5]: new line nag :-)

**Comment by zahidMed** [comment]
Hi @marcingrzejszczak I would like to share with you some remarks regarding the implementation of tracing from my experience in a similar project https://github.com/zahidMed/spring-batch-context which supports other frameworks including Sleuth and I hop this can help:

I am wondering if other teams would like to implement propagation in spring-batch (spring-security for example) by overriding JobBuilderFactory/StepBuilderFactory so developers will be obliged to choose only one of them since only one factory can be provided. Maybe an abstraction layer should be added in order to allow other teams to inject there contexts. In my approach I used aspect to inject context information as job parameters with an abstraction interface that can be implemented according to the context you would like to propagate (class JobExecutionContextListener) .

Another point that I am facing, is that I need to restore the trace context before all the other Step listeners and should be closed after all the other Step listeners so if a developer creates his own step listener, the logs inside his beforeStep() and afterStep() will be traced with the same traceId.

I hope this can help.

File: spring-cloud-sleuth-autoconfigure/pom.xml [hunk 2]
```diff
@@ -83,6 +83,16 @@
...
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-batch</artifactId>
+			<optional>true</optional>
+		</dependency>
+		<dependency>
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<optional>true</optional>
```
[the remark this answers is not in the document]
Comment (marcingrzejszczak) [reply spring-cloud-sleuth-autoconfigure/pom.xml:94]: I should set it in the `test` scope - my bad.

File: tests/brave/spring-cloud-sleuth-instrumentation-batch-tests/pom.xml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			<artifactId>spring-boot-starter-aop</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-batch</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<scope>runtime</scope>
```
[the remark this answers is not in the document]
Comment (marcingrzejszczak) [reply tests/brave/spring-cloud-sleuth-instrumentation-batch-tests/pom.xml:70]: there is no production code here so it doesn't really matter

**Comment by marcingrzejszczak** [comment]
Closing in favour of https://github.com/spring-cloud/spring-cloud-sleuth/pull/1919

**Pull request closed without merging**